app/src/lib/stores/pull-request-coordinator.ts

```diff
@@ -53,10 +51,18 @@ export class PullRequestCoordinator {
   ) {
     // register an update handler for the repositories store
     this.repositoriesStore.onDidUpdate(allRepositories => {
-      this.repositories = allRepositories.filter(
-        isRepositoryWithGitHubRepository
+      this.repositories = Promise.resolve(
+        allRepositories.filter(isRepositoryWithGitHubRepository)
       )
     })
+
+    this.repositories = this.repositoriesStore
+      .getAll()
+      .then(x => x.filter(isRepositoryWithGitHubRepository))
+      .catch(e => {
+        log.error(`PullRequestCoordinator: Error loading repositories`)
+        return []
+      })
```

Comment:
**DISCLAIMER:** this comment is heavily biased by my lack of JS/TS experience and how I used to work with asynchronous code in previous codebases. Consider it just some simple "food for thought" where no actions are actually required.

While I'm amazed by the magic of Promises/Futures 🤩, it feels a bit misleading to me having a property that looks like an immediate value but holds a future value instead (and therefore you need to either `Promise.resolve` when setting immediate values, and `await` when reading its value).

I think my first approach would've been to keep the `repositories` property as an immediate value, and then store it here and then update all the PRs:
```typescript
    this.repositoriesStore
      .getAll()
      .then(x => {
        this.repositories = x.filter(isRepositoryWithGitHubRepository))
      })
      .catch(e => {
        log.error(`PullRequestCoordinator: Error loading repositories`)
        this.repositories = []
      })
      .finally(() => {
        this.handleRepositoriesUpdatedOrSomethingPlz()
      })
```

But of course `handleRepositoriesUpdatedOrSomethingPlz` lacks of some data that the `onPullRequestsChanged` handler has when it's invoked. Trying to compensate/fix that lack of data would involve some extra complexity, so I see how `await` solves the problem in a simpler and more elegant way. My questions are…
1. Is it worth renaming `repositories` to something more explicit of its non-immediate nature? (nope, I can't think of good name suggestions 😂)
1. Is this whole "use what seems to be a regular property as an async value" a common pattern in JS/TS or this codebase? 😳

Reply:
We talked about this in zoom. It's a common practice in JS-land albeit a confusing one to reuse promises. Promises are always eagerly executed so they act as a memoization constructs when multiple consumers await them